dependency-resolution/index.md

```diff
@@ -59,3 +59,25 @@ The rough list is then simplified, by keeping only the newest version of any lis
 Hence, `Dx` which has the highest `semver` version is selected and that is `Dx v1.5.0`.
 
 By following this algorithm, the build list is guaranteed to include the oldest module versions available that meet the requirements, thus achieving reproducible behavior.
+
+## Node.js + npm
+
+[npm](https://docs.npmjs.com/about-npm) is a package manager for [Node.js](https://nodejs.org/en), it facilitates the installation of dependencies for use by Node.js.
+To understand what dependency is used at runtime we need to understand both npm and Node.js work.
+
+As a runtime for an interpreted language, Node.js resolves dependencies at runtime.
+To do this it starts to look for a folder structure matching the dependency name inside a `node_modules/` directory in the directory of the file importing the dependency.
+If it is not found it will repeat this process in the parent folder. This is repeated until, depending on the configuration, some kind of root directory is reached.
+Importantly, this is also how transitive dependencies are resolved.
+
+As a package manager for Node.js, npm installs dependencies by downloading them from the internet (a package registry or source repository) and putting them in the `node_modules/` directory.
+For re-used dependencies two things may happen.
+If a common version is supported by multiple dependencies, npm will try to put the common version directly in the `node_modules/` directory of the parent project, where it can be shared.
```

Comment:
```suggestion
If a common version is supported by multiple dependencies, npm will try to put the common version directly in the `node_modules/` directory of the common parent project of those dependencies, where it can be shared.
```

Reply:
If I'm reading your suggestion correctly it seems you're assuming multiple workspaces? I'd like to keep the description here to projects without workspaces because it's simpler and, I think, naturally extends to workspaces anyway.

Reply:
> you're assuming multiple workspaces

I did not know they were called multiple workspaces. But I agree,we can stick to projects that are not workspaces.